### PR TITLE
[Validator] Wire NotCompromisedPassword in FrameworkBundle and handle non UTF-8 password

### LINKED ISSUE #30870: NotPwnedValidator should convert passwords to UTF-8
Before computing the hashes, we must ensure the passwords are encoded/converted in UTF-8, as expected by the remote DB/API.

### PR DESCRIPTION
**Pull request by tgalopin**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/issues/30870
| License       | MIT
| Doc PR        | -

Live from #eu-fossa

Fix https://github.com/symfony/symfony/issues/30870

### CODE REVIEW

**Review by OskarStark — COMMENTED** [review]
A test case would be nice

**Review by dbrumann — APPROVED** [review]

**Review by ostrolucky — CHANGES_REQUESTED** [review]
I don't think it makes too much sense to make this configurable. People who don't use UTF-8 should change internal_encoding php.ini setting

**Review by maxhelias — APPROVED** [review]

**Review by derrabus — CHANGES_REQUESTED** [review]
Please add a test case.

**Comment by stof** [comment]
FrameworkBundle should define a service for that constraint validator, passing `%kernel.charset%` for this argument. Otherwise, customizing this charset will be a pain in a fullstack project.

**Comment by tgalopin** [comment]
@stof that's the aim, but it can be the subject of another PR IMO.

**Comment by tgalopin** [comment]
Updated

**Review by OskarStark — APPROVED** [review]
Nice one 🎉

**Review by fabpot — APPROVED** [review]

**Comment by tgalopin** [comment]
Based on https://github.com/symfony/symfony/pull/30889, otherwise ready to review.

**Comment by fabpot** [comment]
@tgalopin Can you  rebase?

**Comment by tgalopin** [comment]
Updated

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/validator.xml [hunk 1]
```diff
@@ -61,6 +61,12 @@
             <tag name="validator.constraint_validator" alias="Symfony\Component\Validator\Constraints\EmailValidator" />
         </service>
 
+        <service id="validator.not_compromised" class="Symfony\Component\Validator\Constraints\NotCompromisedPasswordValidator">
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/validator.xml]: validator.not_compromised_paasword ?

Comment (pborreli) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/validator.xml]: not_compromised_password

**Comment by tgalopin** [comment]
Updated

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @tgalopin.

**Pull request merged**